### ISSUE TITLE
hidden type `[async fn body@crates/resource/src/volume/share_fs_volume.rs:50:23: 214:6]` captures lifetime '_#35r

### ISSUE THREAD

**Issue by Apokleos** [kata-containers/kata-containers#7044]
# Description of problem
kata-containers main branch make failed with error:
```
error[E0700]: hidden type for `impl futures::Future<Output = Result<ShareFsVolume, anyhow::Error>>` captures lifetime that does not appear in bounds
   --> crates/resource/src/volume/share_fs_volume.rs:50:23
    |
50  |       ) -> Result<Self> {
    |  _______________________^
51  | |         // The file_name is in the format of "sandbox-{uuid}-{file_name}"
52  | |         let file_name = Path::new(&m.source).file_name().unwrap().to_str().unwrap();
53  | |         let file_name = generate_mount_path("sandbox", file_name);
...   |
213 | |         Ok(volume)
214 | |     }
    | |_____^
    |
    = note: hidden type `[async fn body@crates/resource/src/volume/share_fs_volume.rs:50:23: 214:6]` captures lifetime '_#35r

For more information about this error, try `rustc --explain E0700`.
error: could not compile `resource` due to previous error
warning: build failed, waiting for other jobs to finish...
make: *** [target/x86_64-unknown-linux-gnu/release/containerd-shim-kata-v2] Error 101

```

# Expected result

make ok

# Solution
original: 
```
    pub(crate) async fn new(
        share_fs: &Option<Arc<dyn ShareFs>>,
        m: &oci::Mount,
        cid: &str,
        readonly: bool,
        agent: Arc<dyn Agent>,
    ) -> Result<Self> {
```
corrected
```
    pub(crate) async fn new(
        share_fs: &Option<Arc<dyn ShareFs>>,
        m: &oci::Mount,
        cid: &str,
        readonly: bool,
        agent: &Arc<dyn Agent>,
    ) -> Result<Self> {
```
patch will be pushed later.

**Comment by ananos** [comment]
I believe this is resolved using rust version 1.69.0 -- can you check what you're building against?

**Comment by Apokleos** [comment]
> I believe this is resolved using rust version 1.69.0 -- can you check what you're building against?
> 
Hi, @ananos Sorry for the bug raised. I missed the details of the rust version in versions.yaml. 
Yes, I have test it with it with new version of rust, and it works well.

I will close it.
